src/renderers/WebGLRenderer.js

```diff
@@ -1483,12 +1762,80 @@ function WebGLRenderer( parameters ) {
 
 			} else {
 
-				_this.renderBufferDirect( camera, scene.fog, geometry, material, object, group );
+				var instanceCount = undefined;
+
+				if ( geometry && geometry.isInstancedBufferGeometry && geometry.maxInstancedCount > 0 ) {
+
+					instanceCount = geometry.maxInstancedCount;
+
+				}
+
+				_this.renderBufferDirect( camera, scene.fog, geometry, material, object, group, instanceCount );
+
+			}
+
+			if ( object.onAfterRender ) {
+
+				object.onAfterRender( _this, scene, camera, geometry, material, group );
 
 			}
 
-			object.onAfterRender( _this, scene, camera, geometry, material, group );
+		}
+
+	}
+
+	function renderInstancedGeometryMap ( instancedGeometryMap, scene, camera, overrideMaterial ) {
+
+		for ( var uuid in instancedGeometryMap ) {
+
+			if ( instancedGeometryMap.hasOwnProperty( uuid ) ) {
+
+				var instancedGeometry = instancedGeometryMap[ uuid ];
+				var geometry = instancedGeometry.geometry;
+
+				for (var groupIndex = 0, groupCount = instancedGeometry.groups.length; groupIndex < groupCount; groupIndex++ ) {
+
+					var instancingGroup = instancedGeometry.groups[ groupIndex ];
+
+					var objects = instancingGroup.objects;
+					var material = overrideMaterial ? overrideMaterial : instancingGroup.material;
+
+					if ( objects.length > 0 ) {
+
+						var group = geometry.groups[ groupIndex ] || null;
 
+						for ( var i = 0, l = objects.length; i < l; i++ ) {
+
+							var object = objects[ i ];
+
+							object.modelViewMatrix.multiplyMatrices( camera.matrixWorldInverse, object.matrixWorld );
+							object.normalMatrix.getNormalMatrix( object.modelViewMatrix );
+
+						}
+
+						if ( objects.length < _this.minInstancingBatchSize ) {
+
+							for ( var i = 0, l = objects.length; i < l; i++ ) {
+
+								_this.renderBufferDirect( camera, scene.fog, geometry, material, objects[ i ], group );
```

Comment:
Actually I believe this may also be the same root cause of faces disappearing for certain BufferGeometries when auto-instancing that I mentioned before - e.g. CylinderBufferGeometry defines multiple groups but only the first group ends up being rendered. If I modify this line and 1829 below to pass `null` instead of `group`, the disappearing faces come back. (That's probably not the correct fix though, just supporting evidence that `group` is the issue.)